x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/Version.java

```diff
@@ -117,11 +100,21 @@ static Version extractVersion(URL url) {
         return new Version(ver, hash, maj, min, rev);
     }
 
-    @Override
-    public String toString() {
+    public String versionString() {
         return "v" + version + " [" + hash + "]";
     }
 
+    // This function helps ensure that a client won't attempt to communicate to a server with less features than its own. Since this check
+    // is part of the client's start-up check that might not involve an actual SQL API request, the client has to do a bare version check
+    // as well.
+    public static boolean isServerCompatible(Version server) {
+        // Starting with this version, the compatibility logic moved from the client to the server. Only relevant for 7.x releases.
+        return server.compareTo(Version.V_7_7_0) >= 0
```

Comment:
The V_7_7_0 constant is leaking - encapsulate that into Version (which already has the constant) to something like `Version.hasVersionCompatibility/isCompatible` or something along those lines so the lines becomes:
`return server.isCompatible()`

Reply:
took on your `.hasVersionCompatibility()` suggestion.